src/Elastic.CommonSchema/Serialization/BaseJsonFormatter.Generated.cs

```diff
@@ -5,233 +5,124 @@
-						4 => Read<IDictionary<string, object>>(ref reader, ecsEvent, (b, v) => b.Labels = v),
-						5 => Read<string[]>(ref reader, ecsEvent, (b, v) => b.Tags = v),
-						6 => Read<Agent>(ref reader, ecsEvent, (b, v) => b.Agent = v),
-						7 => Read<As>(ref reader, ecsEvent, (b, v) => b.As = v),
-						8 => Read<Client>(ref reader, ecsEvent, (b, v) => b.Client = v),
-						9 => Read<Cloud>(ref reader, ecsEvent, (b, v) => b.Cloud = v),
-						10 => Read<Container>(ref reader, ecsEvent, (b, v) => b.Container = v),
-						11 => Read<Destination>(ref reader, ecsEvent, (b, v) => b.Destination = v),
-						12 => Read<Dns>(ref reader, ecsEvent, (b, v) => b.Dns = v),
-						13 => Read<Ecs>(ref reader, ecsEvent, (b, v) => b.Ecs = v),
-						14 => Read<Error>(ref reader, ecsEvent, (b, v) => b.Error = v),
-						15 => Read<Event>(ref reader, ecsEvent, (b, v) => b.Event = v),
-						16 => Read<File>(ref reader, ecsEvent, (b, v) => b.File = v),
-						17 => Read<Geo>(ref reader, ecsEvent, (b, v) => b.Geo = v),
-						18 => Read<Group>(ref reader, ecsEvent, (b, v) => b.Group = v),
-						19 => Read<Hash>(ref reader, ecsEvent, (b, v) => b.Hash = v),
-						20 => Read<Host>(ref reader, ecsEvent, (b, v) => b.Host = v),
-						21 => Read<Http>(ref reader, ecsEvent, (b, v) => b.Http = v),
-						22 => Read<Log>(ref reader, ecsEvent, (b, v) => b.Log = v),
-						23 => Read<Network>(ref reader, ecsEvent, (b, v) => b.Network = v),
-						24 => Read<Observer>(ref reader, ecsEvent, (b, v) => b.Observer = v),
-						25 => Read<Organization>(ref reader, ecsEvent, (b, v) => b.Organization = v),
-						26 => Read<Os>(ref reader, ecsEvent, (b, v) => b.Os = v),
-						27 => Read<Package>(ref reader, ecsEvent, (b, v) => b.Package = v),
-						28 => Read<Process>(ref reader, ecsEvent, (b, v) => b.Process = v),
-						29 => Read<Related>(ref reader, ecsEvent, (b, v) => b.Related = v),
-						30 => Read<Server>(ref reader, ecsEvent, (b, v) => b.Server = v),
-						31 => Read<Service>(ref reader, ecsEvent, (b, v) => b.Service = v),
-						32 => Read<Source>(ref reader, ecsEvent, (b, v) => b.Source = v),
-						33 => Read<Threat>(ref reader, ecsEvent, (b, v) => b.Threat = v),
-						34 => Read<Tls>(ref reader, ecsEvent, (b, v) => b.Tls = v),
-						35 => Read<Tracing>(ref reader, ecsEvent, (b, v) => b.Tracing = v),
-						36 => Read<Url>(ref reader, ecsEvent, (b, v) => b.Url = v),
-						37 => Read<User>(ref reader, ecsEvent, (b, v) => b.User = v),
-						38 => Read<UserAgent>(ref reader, ecsEvent, (b, v) => b.UserAgent = v),
-						39 => Read<Vulnerability>(ref reader, ecsEvent, (b, v) => b.Vulnerability = v),
-						_ => false
-					};
-					if (!read)
-					{
-						reader.ReadNext();
-					}
-				}
-				else reader.ReadNext();
-			}
-			ecsEvent.Log ??= new Log();
-			ecsEvent.Log.Level = loglevel;
-			ecsEvent.Message = message;
-			ecsEvent.Timestamp = timestamp;
-			return ecsEvent;
+				"@timestamp" => ReadDateTime(ref reader, ref @timestamp),
```

Comment:
I see this file is autogenerated, but wow the method parameter ended up being `ref @timestamp` - which as it seems does not change the variable name 😅  - so `@timestamp` is the same as `timestamp`. Very funky.

Reply:
Hehe @ allows you to use otherwise reserved keywords as variablenames but can be used in front of all variable names however nonsensical :)